src/accelerate/accelerator.py

```diff
@@ -972,14 +972,14 @@ def trigger_sync_in_backward(model):
             model.require_backward_grad_sync = old_require_backward_grad_sync
             model.require_forward_param_sync = old_require_forward_param_sync
 
-    def _do_sync(self):
+    def _do_sync(self, force: bool = True):
         "Sets the right `sync_gradients` context and either resets or increases `self.step`"
         if self.gradient_state.sync_with_dataloader and self.gradient_state.end_of_dataloader:
             self.step = 0
             self.gradient_state._set_sync_gradients(True)
         else:
             self.step += 1
-            self.gradient_state._set_sync_gradients((self.step % self.gradient_state.num_steps) == 0)
+            self.gradient_state._set_sync_gradients(force or ((self.step % self.gradient_state.num_steps) == 0))
```

Comment:
I don't think there's a reason to have this be a passthrough, Accelerator will always have a gradient_state so we can pull from it

Reply:
@muellerzr Sorry I made a mistake on line 975 where the default value for `force` in `_do_sync(self, force: bool = False)`  should be `False`. Hence if `force=False` it will still check the `((self.step % self.gradient_state.num_steps) == 0` condition.

The rationale for introducing the `force` flag, is so that then `accumulate` might be more explicit. Because then it reads as below, where if `sync_each_batch=False`, then `self.sync_gradients` will be gauranteed to be true and thus the `nullcontext` will be selected. 

```python
@contextmanager
def accumulate(self, *models):

    # sync_each_batch=True will gaurantee below that self.sync_gradients=True, therefore 
    # resulting in the nullcontext always being selected.
    self._do_sync(force=self.gradient_state.plugin_kwargs["sync_each_batch"])
    with contextlib.ExitStack() as cm_stack:
        for m in models:
            cm_stack.enter_context(contextlib.nullcontext() if self.sync_gradients else self.no_sync(m))
        yield
```

However if you prefer, I can move the `self.gradient_state.plugin_kwargs["sync_each_batch"])` check inside `_do_sync` (and in turn removing `force` from the signature of `_do_sync`).  Please let me know which you think is better?

Reply:
Let's just go ahead with it in `_do_sync` then, thanks for the explanation!